galaxy_api/auth/auth.py

```diff
@@ -0,0 +1,109 @@
+import base64
+import json
+
+from django.conf import settings
+from django.db import transaction
+from django.db.utils import IntegrityError
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.permissions import BasePermission
+
+from galaxy_api.auth.models import Tenant, User
+
+
+class RHIdentityAuthentication(BaseAuthentication):
+    """
+    Authenticates users based on RedHat identity header.
+
+    For users logging in first time creates User record and
+    Tenant record for user's account if it doesn't exist.
+    """
+
+    header = 'HTTP_X_RH_IDENTITY'
+
+    def authenticate(self, request):
+        """
+        Authenticates user.
+
+        Raises:
+            AuthenticationFailed: If invalid identity header provided.
+        """
+        if self.header not in request.META:
+            return None
+
+        header = self._decode_header(request.META[self.header])
+
+        try:
+            identity = header['identity']
+            account = identity['account_number']
+
+            user = identity['user']
+            username = user['username']
+        except KeyError:
+            raise AuthenticationFailed
```

Comment:
Worth adding some detail on what AuthenticationFailed?

Or is it left blank without a .authenticate_header() method to cause a 403 without any extra info? (if I'm reading https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication correctly).